### PR TITLE
Get All SFO API added - Scottish-Natural-Heritage/Licensing#716

### PR DESCRIPTION
**Pull request by jmassie**
Added Get All SFO API #716

### CODE REVIEW

**Comment by MikeCoats** [comment]
@jmassie The reason for this first failure...

```
 1.  AssertionError  JSON response length is greater than 1                
                     expected 0 to be above 1                              
                     at assertion:1 in test-script                         
                     inside "Get All Standard Forestry Operations"    
```

...is that your get all call happens before any inserts. You've got a test that says you should return at least two items, but if you start with an empty database (like CI/CD on GA does) then you'll get 0 items in response. You need to add something like...

```js
const port = pm.collectionVariables.get("SFO_API_PORT");
const path = pm.collectionVariables.get("SFO_API_PATH_PREFIX");

for (let i = 0; i < 3; i++) {
  pm.sendRequest({
    url:`http://localhost:${port}/${path}/v1/applications`,
    method:'POST'
  });
}
```

...to the `Pre-req.` tab on your `Get All Standard Forestry Operations` request to make sure that it inserts some rows before it searches for them.

The reason for the second and third failures...

```
2.  AssertionError  Status code is 200 OK                                 
                     expected response to have status code 200 but got 500 
                     at assertion:0 in test-script                         
                     inside "Update an Application"                        
                                                                           
 3.  AssertionError  Returned object matches posted object                 
                     expected false to deeply equal undefined              
                     at assertion:1 in test-script                         
                     inside "Update an Application"         
```

...are because GAs triggered from a PR don't get given any secrets, so it doesn't pass in a `SFO_NOTIFY_API_KEY` and the `PUT` test fails. If you take a look at [this part of my latest TR-API PR](https://github.com/Scottish-Natural-Heritage/trap-registration-api/pull/34/commits/023e289a75e67193f524cedb77fe877dcdb7a948#diff-d7f41ba1aa74f41ef4b0c2337abe593d08092ffae6528a8aa2db712afeee0b66L297-R306) You can see how I've 'hacked' around that.

**Review by MikeCoats — CHANGES_REQUESTED** [review]
The main thing here is to include the Setts when you return the Applications. That will let you search for GridRefs in the front-end.

File: src/controllers/application.js [hunk 1]
```diff
@@ -103,6 +103,15 @@ const ApplicationController = {
     return Application.findByPk(id, {include: Sett});
   },
 
+  /**
+   * Retrieve all applications from the database.
+   *
+   * @returns  {Sequelize.Model} All existing applications.
+   */
+  findAll: async () => {
+    return Application.findAll();
```
[the remark this answers is not in the document]
Comment (jmassie) [reply src/controllers/application.js:112]: Resolved

[the remark this answers is not in the document]
Comment (MikeCoats) [reply src/controllers/application.js]: Resolved in 13e1fc9e3c9f5b9d46f39d2bfe5fe27de3a25d8a.

File: package-lock.json [hunk 2]
```diff
@@ -9457,6 +9487,12 @@
...
         },
+        "lodash": {
+          "version": "4.17.15",
+          "resolved": "https://registry.npmjs.org/lodash/-/lodash-4.17.15.tgz",
+          "integrity": "sha512-8xOcRHvCjnocdS5cpwXQXVzmmh5e5+saE2QGoeQmbKmRS6J3VQppPOIt0MnmE+4xlZoumy0GPG0D0MVIQbNA1A==",
+          "dev": true
+        },
         "uuid": {
           "version": "3.3.2",
           "resolved": "https://registry.npmjs.org/uuid/-/uuid-3.3.2.tgz",
```
[the remark this answers is not in the document]
Comment (MikeCoats) [reply package-lock.json]: Resolved in 92225a06eee7a0caa9f6239da15915522538e723.

File: tests/standard-forestry-operations-api.postman_collection.json [hunk 3]
```diff
@@ -87,13 +85,52 @@
 			},
 			"response": []
 		},
+		{
+			"name": "Get All Standard Forestry Operations",
```
[the remark this answers is not in the document]
Comment (MikeCoats) [reply tests/standard-forestry-operations-api.postman_collection.json]: Resolved in 4d026663425f09aeea9a6748030cb3759ea74922.

**Review by MikeCoats — APPROVED** [review]
This looks good and passes all tests on my local PC.

**Pull request merged**